src/main/java/seedu/address/logic/parser/AlfredParser.java

```diff
@@ -49,6 +55,15 @@ public Command parseCommand(String userInput) throws ParseException {
         case AddParticipantCommand.COMMAND_WORD:
             return new AddParticipantCommandParser().parse(arguments);
 
+        case FindParticipantCommand.COMMAND_WORD:
```

Comment:
I found out that you can use `getPreamble()` to retrieve the entity. Wanna check out my PR #87? It's in ExportCommandParser.java You don't need three parser classes per entity if you do it this way.

Reply:
Thanks for the tip, didnt know about that. 😄

Reply:
Might be wrong here, but I think I still need the 3 parsers because of the single generics. Is it okay if I keep the 3 classes then? I might also extend these classes further to customize the behaviour for find team/ find participant (eg find by their specific fields)

Reply:
You're right, you do still need the 3 parsers, especially if you are going to customise their behaviour.